### DISCUSSION TITLE
Zones Component

### DISCUSSION THREAD

**Discussion by Maksims** [playcanvas/engine#2918] [Ideas]
Currently, there is a Zone Component, which is very minimal and simple, would be good to have an improved version of it.

Here is my use cases that I encounter often working on various projects:
1. Box Projected CubeMaps - setting their box projections, is not easy, as it is just list of numbers, no gizmos are shown for it.
2. Defining zones, for checks if point is in zone, for audio and rendering stuff. We have a project with many rooms, and in order to define zones, we use primitive cube, and have to scale it. Problem is that scaling gizmo is hard to do precise scaling, to match a walls. Zone Component in Editor used to have nice gizmos for dragging walls, which allows to precisely set zones matching rooms.
3. Checking for point in zone. Currently we need to create those primitive cubes, scale them, and ensure they do not overlap, but do touch, which is very hard to do precisely with scale + position + rotation. Then we have to iterate through them, and create OrientedBox for further use. ZoneComponent would have it straight away available.
4. Audio + Zone related behaviours, changing audio parameters based on zones. It would be easy to do with entity that have zone, and a simple script on it with info on reverb for example.

**Comment by MAG-AdrianMeredith** [comment]
You had my hopes up there thinking there was some built in for this behaviour (zone component appears to not be implemented?). Definitely would be really handy as I'm looking to do both light probe volumes and multiple reflection probes

**Comment by mvaligursky** [comment]
There's this PR https://github.com/playcanvas/engine/pull/5123

**Comment by MAG-AdrianMeredith** [comment]
great I'll have a look, don't want to implement this twice If possible
